src/app/app.component.ts

```diff
@@ -23,3 +27,69 @@ export class AppComponent {
 
   }
 }
+
+@Component({
+  selector: 'app-calendar-header',
+  template: `
+    <mat-select placeholder="Calendar Format" [formControl]="form.controls['calendar']">
+      <mat-option *ngFor="let cal of supportedCalendarFormats" [value]="cal">{{cal}}</mat-option>
+    </mat-select>
+    <mat-calendar-header></mat-calendar-header>
+  `,
+  styleUrls: []
+})
+export class HeaderComponent<D> implements OnInit {
+  constructor(@Host() private _calendar: MatCalendar<JDNConvertibleCalendar>,
+              private _dateAdapter: DateAdapter<JDNConvertibleCalendar>,
+              @Inject(FormBuilder) private fb: FormBuilder) {
+  }
+
+  form: FormGroup;
+
+  supportedCalendarFormats = JDNConvertibleCalendar.supportedCalendars;
+
+  activeFormat;
+
+  ngOnInit() {
+
+    this.activeFormat = 'Gregorian';
+
+    if (this._dateAdapter instanceof JDNConvertibleCalendarDateAdapter) {
+      this.activeFormat = this._dateAdapter.activeCalendarFormat;
+    }
+
+    // build a form for the calendar format selection
+    this.form = this.fb.group({
+      calendar: [this.activeFormat, Validators.required]
+    });
+
+    // update the selected calendar format
+    this.form.valueChanges.subscribe((data) => {
+      this.convertCalendar(data.calendar);
+    });
+
+  }
+
+  /**
+   * Converts the date in the current format into the target format.
+   *
+   * @param {"Gregorian" | "Julian"} calendar the target calendar format.
+   */
+  convertCalendar(calendar: 'Gregorian' | 'Julian') {
+
+    if (this._dateAdapter instanceof JDNConvertibleCalendarDateAdapter) {
+
+      const convertedDate = this._dateAdapter.convertCalendarFormat(this._calendar.activeDate, calendar);
+
+      this._calendar.activeDate = convertedDate;
+
+      // update the calendar UI after date format conversion
+      // TODO: today's date is not reset correctly
+      this._calendar.ngAfterContentInit();
```

Comment:
@mmalerba today's date is still not updated correctly (circle in UI). I would be glad for any suggestions.

Reply:
see https://github.com/dhlab-basel/JDNConvertibleCalendarDateAdapter/pull/6#discussion_r182835373

Reply:
I think you need to call init on the various view themselves, rather than the calendar:

```ts
let view = this._calendar.currentView == 'month' ? this._calendar.monthView :
    this._calendar.currentView == 'year ? this._calendar.yearView : this._calendar.multiYearView
this._calendar.monthView.ngAfterContentInit();
this._calendar.yearView.ngAfterContentInit();
this._calendar.multiYearView.ngAfterContentInit();
```
If you want to send a PR adding a `updateTodaysDate()` method to the datepicker, I would be ok with that. This solution seems kind of messy.

Reply:
Thanks, I will try that!